apps/src/lib/kits/maker/boards/microBit/MBFirmataWrapper.js

```diff
@@ -9,14 +10,60 @@ export default class MicrobitFirmataWrapper extends MBFirmataClient {
     this.digitalCallbacks = [];
   }
 
-  connectBoard() {
+  connectBoard(port) {
     return Promise.resolve()
-      .then(() => this.connect())
+      .then(() => this.setSerialPort(port))
       .then(() => {
         return this.setAnalogSamplingInterval(SAMPLE_INTERVAL);
       });
   }
 
+  // Used in setSerialPort
+  dataReceived(data) {
+    if (this.inbufCount + data.length < this.inbuf.length) {
```

Comment:
Out of curiosity, what happens if this condition is not satisfied?

Reply:
This I don't know. My assumption is nothing? I added this function for use in the setSerialPort wrapper and it comes directly from the MBFirmataClient - it got it's own function here rather than being nested in setSerialPort like in the MBFirmataClient because of linting rules. Would it make sense to have clearer comments about this function and it's purpose?

Reply:
I think a brief comment might be good.  I was mainly curious if we knew of any situations in which data would be received but wouldn't be processed because of the buffer length check.